manager/src/main/java/com/redhat/service/bridge/manager/BridgesServiceImpl.java

```diff
@@ -54,6 +70,22 @@ public Bridge createBridge(String customerId, BridgeRequest bridgeRequest) {
         bridge.setSubmittedAt(ZonedDateTime.now(ZoneOffset.UTC));
         bridge.setCustomerId(customerId);
         bridgeDAO.persist(bridge);
+
+        if (rhoasEnabled) {
+            String topicName = String.format("ob-%s", bridge.getId());
+            String serviceAccountName = String.format("ob-%s-consumer", bridge.getId());
+            TopicAndServiceAccountRequest request = new TopicAndServiceAccountRequest(topicName, serviceAccountName);
+            try {
+                rhoasClient.createTopicAndConsumerServiceAccount(request).await().atMost(Duration.ofSeconds(rhoasTimeout));
```

Comment:
@kostola I don't think the user of the `RhoasClient` should have to think about the timeout, otherwise we have to duplicate this logic wherever we use it.

I think it makes more sense that the default timeout is a concern of the `RhoasClient` and it does the waiting on behalf of the user. This means we have centralised control over timeouts when working with other services.

The user then expects the `RhoasClient` to return either the successful `TopicAndServiceAccountResponse` or throw an `InternalPlatformException` if something went wrong. That would mean a signature on the RhoasClient of something like:

```
TopicAndServiceAccountResponse createTopicAndConsumerServiceAccount(TopicAndServiceAccountRequest r);
```

Reply:
In this way the `RhoasClient` would become blocking, losing the advantage of the async implementation. Are we going to stick with thread-per-request model for handling HTTP request?

Reply:
@kostola OK, let's leave it as it is for now, but can you please set the `CompletionException` and `TimeoutException` as the `cause` on an  `InternalServiceException` with the log statement as the message. At the minute we don't have a non-blocking use-case on the client (we need the topic + service account to be created before the Shard starts to try and deploy the OpenBridge instance), but I think that may change as we progress the implementation.

What we need to think about is "what happens if the Fleet Manager crashes halfway through creating the topic". How would we recover that scenario and continue with the topic creation in an idempotent manner? Lets get this PR resolved and merged, and then we can start to consider how we approach that problem at a later date.

Reply:
Done :+1: